spectral_cube/spectral_cube.py

```diff
@@ -2363,6 +2363,7 @@ def __init__(self, data, wcs, mask=None, meta=None, fill_value=np.nan,
             self.pixels_per_beam = (self.beam.sr /
                                     (astropy.wcs.utils.proj_plane_pixel_area(self.wcs) *
                                      u.deg**2)).to(u.dimensionless_unscaled).value
+        self.external_update_function = None  # External function that updates a gui's progress bar
```

Comment:
Again, why not make this a function parameter instead of a class attribute?

Reply:
actually there is no need for that on my end. I can make this a parameter.